File: 67. Add Binary.go
func addBinary(a string, b string) string {
    ans := ""
    i := len(a) - 1
    j := len(b) - 1
    carry := 0
    for i >= 0 || j >= 0 {
        sum := carry
        if i >= 0 {
            sum += int(a[i]) - '0'
            i -= 1
        } else {
            sum += 0
        }
        if j >= 0 {
            sum += int(b[j]) - '0'
            j -= 1
        } else {
            sum += 0
        }
        ans = strconv.Itoa(sum%2) + ans
        carry = sum / 2
    }
    if carry == 1 {
        ans = strconv.Itoa(1) + ans
    }
    return ans
}

File: 67. Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans="";
        int i=a.length()-1, j=b.length()-1, carry=0, sum;
        while(i>=0||j>=0){
            sum=carry;
            sum+=(i>=0)?a[i--]-'0':0;
            sum+=(j>=0)?b[j--]-'0':0;
            ans=to_string(sum%2)+ans;
            carry=sum/2;
        }
        if(carry)   ans=to_string(1)+ans;
        return ans;
    }
};